crates/ra_hir_ty/src/db.rs

```diff
@@ -22,6 +22,7 @@ use hir_expand::name::Name;
 #[salsa::requires(salsa::Database)]
 pub trait HirDatabase: DefDatabase + Upcast<dyn DefDatabase> {
     #[salsa::invoke(infer_wait)]
+    #[salsa::transparent]
     fn infer(&self, def: DefWithBodyId) -> Arc<InferenceResult>;
```

Comment:
@matklad, what is the rationale for `infer_wait`? And am I right that that there is even no need for `salsa::transparent` here, since we could just do a default impl 
```rust
fn infer(...) {
    infer_wait(...);
}
```

Reply:
There are no docs on `salsa::transparent`, so I am wondering what it does